pkg/controllers/dnspolicy/dnspolicy_controller.go

```diff
@@ -194,6 +217,16 @@ func (r *DNSPolicyReconciler) deleteResources(ctx context.Context, dnsPolicy *v1
 		}
 	}
 
+	gatewayDiffObj, err = r.ComputeGatewayDiffs(ctx, dnsPolicy, targetNetworkObject, &DNSPolicyRefsConfig{})
+	if err != nil {
+		return err
+	}
```

Comment:
Did you need to call this (r.ComputeGatewayDiffs) again?

Reply:
hmm not sure. It worked as expected?